File: go/string/824. 山羊拉丁文.go
package string

import (
	"strings"
)

func toGoatLatin(S string) string {
	res := strings.Split(S, " ")
	for i, v := range res {
		b := []byte(v)
		switch b[0] {
		case 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U':
			res[i] += "ma"
			res[i] += strings.Repeat("a", i+1)
		default:
			res[i] = res[i][1:] + res[i][0:1] + "ma"
			res[i] += strings.Repeat("a", i+1)
		}
	}
	return strings.Join(res, " ")
}

File: python/字符串/824. 山羊拉丁文.py
class Solution:
    def toGoatLatin(self, S: str) -> str:
        res = S.split(" ")
        for i,word in enumerate(res):
            if word[0].lower() in "aeiou":
                res[i] += "ma"
                res[i] += "a"*(i+1)
            else:
                res[i] = word[1:]+word[0]+"ma"
                res[i] += "a"*(i+1)
        return " ".join(res)
